data/thermoml_archive/transform.py

```diff
@@ -0,0 +1,140 @@
+            ):
+                f.write(chunk)
+
+    # check if checksum is correct
+    sha256 = hashlib.sha256()
+    with open(download_path, "rb") as f:
+        for chunk in tqdm.tqdm(iter(lambda: f.read(8192), b""), desc="Checking hash"):
+            sha256.update(chunk)
+
+    if received_hash := sha256.hexdigest() != sha256_checksum:
+        raise RuntimeError(
+            "Downloaded file did not match expected checksum -- "
+            "either a new version has been released or something has gone wrong!\n"
+            f"Expected: {sha256_checksum}\n"
+            f"Received: {received_hash}"
+        )
+
+    # Extract tar.gz archive
+    with tarfile.open(download_path, "r:*") as tar:
+        tar.extractall(pathlib.Path(__file__).parent)
+
+    # Loop through journal DOI folders and scrape files
+
+    if final_csv_path.exists():
+        sha256 = hashlib.sha256()
+        with open(final_csv_path, "rb") as f:
+            for chunk in tqdm.tqdm(
+                iter(lambda: f.read(8192), b""), desc="Checking hash"
+            ):
+                sha256.update(chunk)
+        if sha256.hexdigest() != final_expected_csv_checksum:
+            warnings.warn(
+                "Old CSV file did not match expected checksum, will try to recreate."
+            )
+        final_csv_path.rename(final_csv_path.with_suffix(".old.csv"))
+
+    root_dois = ("10.1007", "10.1016", "10.1021")
+
+    num_points = 0
+    num_failed = 0
+    for doi in root_dois:
+        for path in tqdm.tqdm(
+            (pathlib.Path(__file__).parent / doi).glob("*.xml"),
+            desc=f"Looping over files in {doi}",
+        ):
+            with open(path, "r") as f:
+                try:
+                    pd.DataFrame(Parser(path).parse()).to_csv(final_csv_path, mode="a")
+                    num_points += 1
+                except Exception:
+                    num_failed += 1
+
+    print(f"Ingested {num_points} with {num_failed} failures.")
+
+    sha256 = hashlib.sha256()
+    with open(final_csv_path, "rb") as f:
+        for chunk in tqdm.tqdm(iter(lambda: f.read(8192), b""), desc="Checking hash"):
+            sha256.update(chunk)
+
+    if csv_hash := sha256.hexdigest() != final_expected_csv_checksum:
+        warnings.warn(
+            "Final CSV file did not match expected checksum!\n"
+            f"Expected: {final_expected_csv_checksum}\n"
+            f"Received: {csv_hash}"
+        )
+
+    # create metadata
+    meta = Dataset(
+        **{
+            "name": "thermoml_archive",
+            "description": "ThermoML is an XML-based IUPAC standard for the storage and exchange of experimental thermophysical and thermochemical property data. The ThermoML archive is a subset of Thermodynamics Research Center (TRC) data holdings corresponding to cooperation between NIST TRC and five journals.",  # noqa
+            "identifiers": [
+                {
+                    "id": "",
```

Comment:
is it ok to not provide an id value?

Reply:
No, this is unfinished (as mentioned in PR desc), this is the main bit that needs feedback/further discussion.